drivers/adodb-text.inc.php

```diff
@@ -176,32 +179,37 @@ function Connect(&$array, $types = false, $colnames = false)
 
 	// returns queryID or false
 	// We presume that the select statement is on the same table (what else?),
-	// with the only difference being the order by.
-	//You can filter by using $eval and each clause is stored in $arr .eg. $arr[1] == 'name'
+	// with the only difference being the ORDER BY.
+	// You can filter by using $eval and each clause is stored in $arr .eg. $arr[1] == 'name'
 	// also supports SELECT [DISTINCT] COL FROM ... -- only 1 col supported
-	function _query($sql,$input_arr,$eval=false)
+	function _query($sql, $input_arr, $eval=false)
 	{
 		if ($this->_origarray === false) return false;
 
-		$eval = $this->evalAll;
+		if (!is_string($eval)) {
+			// This looks like a workaround for ADOdb 5 not having a 3rd execute() parameter by setting
+			// that evalAll object variable before calling execute()
+			$eval = $this->evalAll;
```

Comment:
The _evalAll_ property is actually not defined anywhere.

The _ADOConnection_ class does define *$_evalAll*, but it is not used anywhere so maybe it should be removed from that class and defined in *ADODB_text* instead ? 

In any case, the property's name should be consistent.

Reply:
Yeah, I found this $_evalAll defined later too in the ADOConnection class and wanted ask that question about too. I'm fine with the removal of it from the ADOConnection class and keep that hacky stuff in the ADODB_text class.

Reply:
> and automatically clear it after query execution ?

Yes, I thought also about it, but was unsure what the original author had in mind with setting and use of the 'evalAll' variable.